Sources/FoundationInternationalization/Formatting/Number/BinaryInteger+FormatStyle.swift

```diff
@@ -43,250 +43,183 @@ extension BinaryInteger {
-    ///
-    /// Note that 'maximum' in this case is context-specific to the `numericStringRepresentation` algorithm.  It is not necessarily the maximum digits required for _any_ Word, but rather any value of Word type which is less than the maximum decimal magnitude.  Typically this is one digit less.
-    internal // For unit test accessibility, otherwise would be fileprivate.
-    static func decimalDigitsAndMagnitudePerWord() -> (digits: Int, magnitude: Self) {
-        // This method cannot be defined statically because it depends on the types of both Self and Word.  The compiler can in principle fold this down to the resulting values at compile time - since it knows the concrete types for any given call site - and then just inline those into the caller.
-
-        // First, a fast-path that works for any type (for `Self`) which can (essentially) represent a UInt (or larger).
-        let guessDigits = Int(Double(Words.Element.bitWidth) * log10(2))
-        let guessMagnitude = pow(10, Double(guessDigits))
-
-        if let magnitudeAsSelf = Self(exactly: guessMagnitude) {
-            return (guessDigits, magnitudeAsSelf)
-        }
-
-        // Alas `Self` is smaller than UInt, so fall back to a truly generic - but slower - algorithm to find the results.  This is because BinaryIntegers - unlike e.g. FixedWidthIntegers - don't provide APIs for questions like "what is the maximum bit width of Self?".
-
-        var count = 1
-        var value: Words.Element = 1
-
-        while true {
-            var (nextValue, overflowed) = value.multipliedReportingOverflow(by: 10)
-
-            // Words.Element might be wider than the actual type, e.g. if Self is UInt8 (and UInt is not).  The magnitude is limited by the smallest of the two.
-            if !overflowed && nil == Self(exactly: nextValue) {
-                overflowed = true
-            }
-
-            if !overflowed || .zero == nextValue {
-                count += 1
-            }
-
-            if overflowed {
-                return (count - 1, Self(value))
-            }
+/// Writes `word` in "Numeric string" format (https://speleotrove.com/decimal/daconvs.html)  to
+/// the trailing edge of `buffer`, then returns the least significant `buffer` index written to.
+///
+/// - Parameters:
+///   - word: An unsigned binary integer's only word.
+///   - buffer: A buffer with enough memory to accommodate the format.
+///
+private func formTrailingNumericStringRepresentationForUnsignedInteger(
+word: UInt, into buffer: Slice<UnsafeMutableBufferPointer<UInt8>>) -> Int {
+    var magnitude = word
+    var index = buffer.endIndex
+    
+    repeat {
+        precondition(index > buffer.startIndex)
+        buffer.base.formIndex(before: &index)
+        
+        let remainder: UInt
+        (magnitude, remainder) = magnitude.quotientAndRemainder(dividingBy: 10)
+        buffer.base[index] = UInt8(ascii: "0") &+ UInt8(truncatingIfNeeded: remainder)
+    }   while magnitude != 0 as UInt
```

Comment:
Is the `as UInt` necessary?  `magnitude` is defined as `Uint` (since it's initialised from `word` which is `UInt`), so there should be no ambiguity nor any significant type inference cost.

Reply:
Dunno. Swift's BinaryInteger protocol comes with has generic comparison operators, Swift's preferred integer literal is Int, and Xcode does not show what integer literals resolve to.

Reply:
I see that  puts a lot of emphesis on avoiding type assumptions. My concern was more about the expression resolving to an Int-to-UInt conversion. In any case, I've changed it to `.zero` since Xcode shows it resolving to the correct type.